resources/fetchers/ecr_fetcher.go

```diff
@@ -20,26 +20,31 @@ package fetchers
 import (
 	"context"
 	"fmt"
+	"github.com/elastic/cloudbeat/resources/providers/awslib"
+	v1 "k8s.io/api/core/v1"
 	"regexp"
 
 	"github.com/aws/aws-sdk-go-v2/service/ecr"
 	"github.com/elastic/beats/v7/libbeat/logp"
 	"github.com/elastic/cloudbeat/resources/fetching"
-	"github.com/elastic/cloudbeat/resources/providers/awslib"
 	"github.com/gofrs/uuid"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	k8s "k8s.io/client-go/kubernetes"
 )
 
-const PrivateRepoRegexTemplate = "^%s\\.dkr\\.ecr\\.%s\\.amazonaws\\.com\\/([-\\w]+)[:,@]?"
-const PublicRepoRegex = "public\\.ecr\\.aws\\/\\w+\\/([\\w-]+)\\:?"
+const PrivateRepoRegexTemplate = "^%s\\.dkr\\.ecr\\.%s\\.amazonaws\\.com\\/([-\\w\\.\\/]+)[:,@]?"
+const PublicRepoRegex = "public\\.ecr\\.aws\\/\\w+\\/([-\\w\\.\\/]+)\\:?"
 
 type ECRFetcher struct {
-	log               *logp.Logger
-	cfg               ECRFetcherConfig
-	ecrProvider       awslib.EcrRepositoryDescriber
-	kubeClient        k8s.Interface
-	repoRegexMatchers []*regexp.Regexp
+	log           *logp.Logger
+	cfg           ECRFetcherConfig
+	kubeClient    k8s.Interface
+	PodDescribers []PodDescriber
+}
+
+type PodDescriber struct {
```

Comment:
Not a must, `PartialPodDescriber`

Reply:
Not sure what you meant by `partial` :/

I kept the name as it was for now